sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/Iso8601DateDeserializer.java

```diff
@@ -9,58 +9,42 @@
 import com.fasterxml.jackson.databind.deser.std.UntypedObjectDeserializer;
 
 import java.io.IOException;
-import java.time.OffsetDateTime;
-import java.time.format.DateTimeFormatter;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
 import java.util.List;
 import java.util.stream.Collectors;
 
-/**
- * Custom deserializer to deserialize strings as instances of {@link OffsetDateTime}
- */
-final class Iso8601DateDeserializer extends UntypedObjectDeserializer {
+public class Iso8601DateDeserializer extends UntypedObjectDeserializer {
     private static final long serialVersionUID = 1L;
     private final UntypedObjectDeserializer defaultDeserializer;
 
-    /**
-     * Constructor
-     *
-     * @param defaultDeserializer the deserializer to use when an OffsetDateTime match is not found
-     */
-    Iso8601DateDeserializer(UntypedObjectDeserializer defaultDeserializer) {
+    protected Iso8601DateDeserializer(final UntypedObjectDeserializer defaultDeserializer) {
         super(null, null);
         this.defaultDeserializer = defaultDeserializer;
     }
 
     @Override
     public Object deserialize(JsonParser jp, DeserializationContext ctxt) throws IOException {
-        if (jp.getCurrentTokenId() == JsonTokenId.ID_STRING) {
-            String value = jp.getText();
-            return parseOffsetDateTime(value);
-        } else if (jp.getCurrentTokenId() == JsonTokenId.ID_START_ARRAY) {
-            List<?> list = (List) defaultDeserializer.deserialize(jp, ctxt);
+        Object obj = defaultDeserializer.deserialize(jp, ctxt);
```

Comment:
Should this be a common class in azure-core? Iso8601 seems like a common format to deserialize?

Reply:
DateTimeSerializer is supposed to deserialize and deserialize this format.. should the fix be in there?
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/DateTimeSerializer.java#L18

Reply:
Search allows to upload user-defined documents with any date classes. 
The link you have is to support Joda time (OffsetDateTime, LocalDateTime etc).
What we are trying to do here is to support old date lib (java.util.Date)
We agreed to put serializer to search only, since Search has specific format and time zone requirement.